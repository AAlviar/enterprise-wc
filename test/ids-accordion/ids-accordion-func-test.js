/**
 * @jest-environment jsdom
 */
import IdsAccordion from '../../src/ids-accordion/ids-accordion';

describe('IdsAccordion Component', () => {
  let el;

  beforeEach(async () => {
    const elem = new IdsAccordion();

    document.body.appendChild(elem);
    el = document.querySelector('ids-accordion');
  });

  afterEach(async () => {
    document.body.innerHTML = '';
    el = null;
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    el.remove();
    el = new IdsAccordion();
    document.body.appendChild(el);

    expect(document.querySelectorAll('ids-accordion').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });
});
