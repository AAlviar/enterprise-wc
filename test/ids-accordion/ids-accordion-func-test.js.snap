// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsAccordion Component renders correctly 1`] = `"<ids-accordion><ids-accordion-panel><ids-accordion-header></ids-accordion-header></ids-accordion-panel><ids-accordion-panel><ids-accordion-header></ids-accordion-header></ids-accordion-panel><ids-accordion-panel><ids-accordion-header></ids-accordion-header></ids-accordion-panel></ids-accordion>"`;

exports[`IdsAccordion Component renders correctly 2`] = `"<ids-accordion><ids-accordion-panel expanded=\\"true\\"><ids-accordion-header></ids-accordion-header></ids-accordion-panel><ids-accordion-panel><ids-accordion-header></ids-accordion-header></ids-accordion-panel><ids-accordion-panel><ids-accordion-header></ids-accordion-header></ids-accordion-panel></ids-accordion>"`;

exports[`IdsAccordion Component renders correctly 3`] = `"<ids-accordion><ids-accordion-panel expanded=\\"false\\"><ids-accordion-header></ids-accordion-header></ids-accordion-panel><ids-accordion-panel><ids-accordion-header></ids-accordion-header></ids-accordion-panel><ids-accordion-panel><ids-accordion-header></ids-accordion-header></ids-accordion-panel></ids-accordion>"`;
